lib/HLSL/DxilLoopDeletion.cpp

```diff
@@ -48,7 +53,7 @@ bool DxilLoopDeletion::runOnFunction(Function &F) {
   legacy::FunctionPassManager SimplifyPM(F.getParent());
   SimplifyPM.add(createCFGSimplificationPass());
   SimplifyPM.add(createDeadCodeEliminationPass());
-  SimplifyPM.add(createInstructionCombiningPass());
+  SimplifyPM.add(createInstructionCombiningPass(/*HLSL No sink*/m_HLSLNoSink));
```

Comment:
Do we need test coverage for this? Is there a case where instcombine can run in the dead loop pass and sink things when `-opt-disable sink` is given?

Or is this change just needed because instcombine takes an argument now?

Reply:
Yes good point. It did sink things in here too. I'll add a test for it.

Reply:
Added a test. Also realized the options are not properly hooked up to -Odump. Fixed it and added a test.